### DISCUSSION TITLE
[FIXED] 'MODULE_NOT_FOUND' error when running any npm command

### DISCUSSION THREAD

**Discussion by Ahmed-A-Khamis** [community/community#30706] [Other Feature Feedback, Questions, & Ideas] [answered]
I'm learning JavaScript with "Morten Rand-Hendriksen" on [LinkedIn Learning](https://www.linkedin.com/learning/javascript-essential-training).
And the instructor said we need to install "Prettier" on our VSCode. He said that after installing it from the "Extensions" in VSCode, we will open the exercise folder on VSCode and type in the terminal `npm install` but when I do so this message appears.
I'm using Windows 10 OS.

```
PS C:\Users\ytrip\OneDrive\Desktop\javascript-essential-training-2832077-main> npm -v
node:internal/modules/cjs/loader:364
      const err = new Error(
                  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\node_modules\are-we-there-yet\node_modules\readable-stream\readable.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:364:19)
    at Function.Module._findPath (node:internal/modules/cjs/loader:577:18)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:942:27)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\are-we-there-yet\lib\tracker-stream.js:3:14)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32) {
  code: 'MODULE_NOT_FOUND',
  path: 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\are-we-there-yet\\node_modules\\readable-stream\\package.json',   
  requestPath: 'readable-stream'
}
node:internal/modules/cjs/loader:364
      const err = new Error(
                  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\node_modules\are-we-there-yet\node_modules\readable-stream\readable.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:364:19)
    at Function.Module._findPath (node:internal/modules/cjs/loader:577:18)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:942:27)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\are-we-there-yet\lib\tracker-stream.js:3:14)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32) {
  code: 'MODULE_NOT_FOUND',
  path: 'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\are-we-there-yet\\node_modules\\readable-stream\\package.json',   
  requestPath: 'readable-stream'
}
```

I posted a question on StackOverFlow and I checked the path in environment variables. I also tried to reinstall nodeJS. I also tried to delete my local node_modules folder. I tried running this on my terminal `SET PATH=C:\Program Files\Nodejs;%PATH%
`. I tried restarting my computer with every change I made.  But everything just seems to not work.

I'm new in this field and I really couldn't find any solution for this problem. And unfortunately, no help is being provided in the discussion forum on LinkedIn Learning. So I'd appreciate your help.

*package.json*:
```lang-js
{
  "name": "javascript-essential-training-exercise-files",
  "version": "1.0.0",
  "description": "Repository for exercise files used in the LinkedIn Learning course `JavaScript Essential Training` instructed by Morten Rand-Hendriksen and published 2020.",
  "main": "index.html",
  "author": "Morten Rand-Hendriksen",
  "license": "SEE LICENSE IN LICENSE",
  "devDependencies": {
    "eslint": "^7.11.0",
    "eslint-config-standard": "^14.1.1",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-standard": "^4.0.1",
    "prettier": "^2.1.2"
  }
} 
```

**Comment by Ahmed-A-Khamis** [comment, marked as answer]
So, despite all the suggested solves I received from StackOverFLow and npm GitHub Support and all the search I've done, none of it worked for me, so I started trying to solve it myself, and fortunately, I HAVE FOUND A FIX FOR THIS ISSUE.

So the problem appears because `npm` terminal is trying to find the files in directory `C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\`   If you notice, the sub directory `node_modules\npm\bin`  is repeated twice even though the actual correct directory is `C:\Program Files\nodejs\node_modules\npm\bin`, and that's what was causing the error `MODULE_NOT_FOUND` because it can not FIND the modules.

So in order to solve this problem, I made a new folder inside the `bin` folder and copied the `node_modules` folder that `npm` was looking for in that directory. And that eventually solved the problem. I don't know why `npm` is trying to search for the modules in that directory but, I hope it gets fixed by npm as I shared that conclusion with them.

Hope this helps anyone who's trying to learn like me but faces this problem.

**Comment by ghost** [comment]
I had same kind of problem but with linux.  Removed node-modules and package-lock.json. -->Updating the nodejs version with nvm  --> running "npm install "again and it found the modules.

**Reply by vungochieu491999** [reply]
Wow. Thanks!

**Comment by tony9823** [comment]
Why does the error appear when I run npm install in the terminal after installing Prettier and opening the exercise folder in VSCode on my Windows 10 machine while following Morten Rand-Hendriksen's LinkedIn Learning course? [Click here](https://apkprequel.com/prequel-old-versions/) for more details

**Comment by Ratnakar-7** [comment]
If you're facing MODULE_NOT_FOUND errors or npm misbehaving in PowerShell, CMD, or VS Code — but working fine in Git Bash — the issue may be due to conflicting or duplicated Node.js paths in your system environment variables.
**Remove all Node.js-related paths from your system variables except: C:\Program Files\nodejs**
No need to manually edit or copy folders. Just clean up the PATH and restart your terminal.

**Comment by chineduojukwu03** [comment]
i am have this 
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\Admin\AppData\Local\npm-cache\_npx\d66c5096c7023bfb\node_modules\commander\esm.mjs' imported from C:\Users\Admin\AppData\Local\npm-cache\_npx\d66c5096c7023bfb\node_modules\shadcn\dist\index.js
Did you mean to import "commander/index.js"?
    at finalizeResolution (node:internal/modules/esm/resolve:274:11)
    at moduleResolve (node:internal/modules/esm/resolve:864:10)
    at defaultResolve (node:internal/modules/esm/resolve:990:11)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:737:20)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:714:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:293:38)
    at #link (node:internal/modules/esm/module_job:208:49) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///C:/Users/Admin/AppData/Local/npm-cache/_npx/d66c5096c7023bfb/node_modules/commander/esm.mjs'
}

Node.js v24.11.1

i try running npx shadcn@latest add dialog
pls i can i fix it

**Comment by zeahan361** [comment]
https://brawlapk.com/, https://maternitypaycalculator.com/, https://writezone.xyz/, https://russella.co.uk/, https://www.minecraftapkx.com/, https://www.chillzonemovies.com/, https://btsworldtours.com/
